wikidatarefisland/pumps/pump.py

```diff
@@ -18,10 +19,26 @@ def run(self, pipe: AbstractPipe, input_file_path: str, output_file_path: str):
         for line in self.storage.getLines(input_file_path):
             output = pipe.flow(line)
             for single_output in output:
-                # TODO: we're opening and closing this on every single line of output
                 self.storage.append(output_file_path, single_output)
 
 
+class DumpReaderPump(AbstractPump):
+    def __init__(self, storage: Storage, batch_size: int):
+        self.storage = storage
+        self.batch_size = batch_size
+
+    def run(self, pipe: AbstractPipe, input_file_path: str, output_file_path: str):
+        values = []
+        for line in self.storage.get_dump_lines(input_file_path):
+            output = pipe.flow(line)
+            for single_output in output:
+                values.append(json.dumps(single_output, ensure_ascii=False))
```

Comment:
I also had a rough stare at if batching helped. With a test file of 2.2G uncompressed (and 203M bz2) I actually saw no noticeable different in write time (I saw +- 10% on repeated runs though) but that could just be due to my machine or the fact I'm not writing to NFS or something.

I tested by using batch_size=1 (of course this still batches the writing from one "flow" of the pipe.

I also tried "streaming" the output without opening and closing the file and using jsonl (see: https://github.com/wmde/reference-island/tree/better_read_jsonl) This also made no noticable difference in runtime.

I finally run cProfile because I was convinced I was missing something. See the attached files on phab:
reference-island3.pstat is parsing bzip
reference-island4.pstat is parsing uncompressed json

Looks like the slowest bits (by far) are decompressing bzip, parsing json and then "property_id_statement_includer".

Reply:
Have you seen https://phabricator.wikimedia.org/T251863#6164580?

Reply:
I did see! Marking as resolved to try and keep github more readable